include/zephyr/net/openthread.h

```diff
@@ -111,23 +165,44 @@ struct openthread_state_changed_cb {
 };
 
 /**
+ * @brief Register callbacks that will be called when a certain configuration
+ * or state changes occur within OpenThread.
+ *
+ * @param cb Callback struct to register.
+ */
+int openthread_state_change_callback_register(struct openthread_state_changed_callback *cb);
```

Comment:
nit: The callback type is `...changed_callback` and the function is `...change_callback` (without `d`). Worth unifying :)